pkg/coscheduling/coscheduling.go

```diff
@@ -95,295 +84,374 @@ func New(obj runtime.Object, handle framework.FrameworkHandle) (framework.Plugin
+	cs.pgMgr.PostBind(ctx, pod, nodeName)
+}
+
+// VerifyLocalCache verify pgStatustatus cache
+func (cs *Coscheduling) VerifyLocalCache(toSchedulePGName string) bool {
+	pgStatus := cs.pgMgr.PodGroupStatusCache().Get(toSchedulePGName)
+	podInCacheUpdate := false
+	// verify pods cache
+	cs.frameworkHandler.IterateOverWaitingPods(func(waitingPod framework.WaitingPod) {
+		klog.V(5).Info("Start IterateOverWaitingPods")
+		pod := waitingPod.GetPod()
+		pgName := core.GetNamedspacedName(pgStatus.PodGroup)
+		klog.V(5).Infof("Start walking through pod %v uid: %v", core.GetNamedspacedName(pod), pod.UID)
+		if pgName == toSchedulePGName {
+			klog.V(5).Infof("Wanted cache pod %v", core.GetNamedspacedName(pod))
+			_, found := pgStatus.AssumedPods.Get(core.GetNamedspacedName(pod))
+			if !found {
+				podInCacheUpdate = true
+				return
+			}
+			klog.V(6).Infof("Current pod group state %+v", pgStatus.AssumedPods.Items())
+			cs.pgMgr.PodGroupStatusCache().Set(core.GetNamedspacedName(pgStatus.PodGroup), pgStatus)
 		}
 	})
+	return podInCacheUpdate
 }
 
-// GetPodGroupLabels checks if the pod belongs to a PodGroup. If so, it will return the
-// podGroupName, minAvailable of the PodGroup. If not, it will return "" and 0.
-func GetPodGroupLabels(pod *v1.Pod) (string, int, error) {
-	podGroupName, exist := pod.Labels[PodGroupName]
-	if !exist || len(podGroupName) == 0 {
-		return "", 0, nil
-	}
-	minAvailable, exist := pod.Labels[PodGroupMinAvailable]
-	if !exist || len(minAvailable) == 0 {
-		return "", 0, nil
+// StartBatchSchedule receives the group name of pod to start scheduling
+func (cs *Coscheduling) StartBatchSchedule(fullName string) {
+	pgStatus := cs.pgMgr.PodGroupStatusCache().Get(fullName)
+	if pgStatus.PodGroup.Status.Phase != v1alpha1.PodGroupPreScheduling && pgStatus.PodGroup.Status.Phase != v1alpha1.
+		PodGroupScheduling {
+		return
 	}
-	minNum, err := strconv.Atoi(minAvailable)
+
+	// record time to avoid abnormal exit when bind
+	pg := pgStatus.PodGroup
+	pgCopy := pgStatus.PodGroup.DeepCopy()
+	pgCopy.Status.ScheduleStartTime = metav1.Now()
+
+	patch, err := util.CreateMergePatch(pg, pgCopy)
 	if err != nil {
-		klog.Errorf("PodGroup %v/%v : PodGroupMinAvailable %v is invalid", pod.Namespace, pod.Name, minAvailable)
-		return "", 0, err
+		return
 	}
-	if minNum < 1 {
-		klog.Errorf("PodGroup %v/%v : PodGroupMinAvailable %v is less than 1", pod.Namespace, pod.Name, minAvailable)
-		return "", 0, err
+
+	if err := cs.pgMgr.PatchPodGroup(pg.Name, pg.Namespace, patch); err != nil {
+		return
+	}
+
+	// Start coscheduling
+	klog.V(4).Infof("Start scheduling %v", fullName)
+	localWaitingPods := cs.pgMgr.GetPodNodePairs(fullName)
+	if localWaitingPods == nil {
+		klog.V(4).Infof("Can not found pending pods for %v", fullName)
+		return
+	}
+	pendingPodNameIDs := cs.pgMgr.GetAssumedPods(fullName)
+	if pendingPodNameIDs == nil {
+		klog.V(4).Infof("Can not found pending IDs for %v", fullName)
+		return
+	}
```

Comment:
This check is redundant right? As it works the same as `localWaitingPods `?

Reply:
Yes. thanks

Reply:
Yes, removed thanks